src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java

```diff
@@ -123,4 +140,45 @@ private static boolean checkEventFileLine(String line, String target) {
     }
     return currentToken.equals(target);
   }
+
+  /**
+   * Get the Export Block Descriptor request for Event File lines after the timestamp argument.
+   *
+   * @param timestampMilliseconds - the start time for request
+   * @return request string
+   */
+  private static String getEbdEventFileSinceAbsoluteTimeRequest(long timestampMilliseconds) {
+    Date TimeDateObj = new Date(timestampMilliseconds);
+    SimpleDateFormat sf = new SimpleDateFormat(FLEXY_EBD_EVENT_FILE_DATE_FORMAT);
+    return EBD_EVENT_LOG_ABSOLUTE_START_BASE + sf.format(TimeDateObj);
+  }
+
+  /**
+   * Read Event File via EBD using absolute start time, check for event Id
+   *
+   * @param eventId - ID of the target event
+   * @param absoluteStartTimeMilliseconds - epoch time in milliseconds as start time
+   * @return boolean - true/false found event ID in logs
+   * @throws EbdTimeoutException - for timeout of EBD request
+   * @throws IOException - for exception with input stream
+   */
+  public static boolean didEventOccurSinceAbsolute(
+      String eventId, long absoluteStartTimeMilliseconds) throws IOException, EbdTimeoutException {
+    Exporter ex =
+        HistoricalDataManager.executeEbdCall(
+            getEbdEventFileSinceAbsoluteTimeRequest(absoluteStartTimeMilliseconds));
+    String exporterFile = StringUtils.getStringFromInputStream(ex, EVENT_FILE_STRING_ENCODING);
+    ex.close();
+
+    if (exporterFile.length() > 0) {
+      final List eventFileLines = StringUtils.split(exporterFile, EVENT_FILE_LINE_DELIMITER);
+      for (int x = 1; x < eventFileLines.size(); x++) {
+        String line = (String) eventFileLines.get(x);
+        if (checkEventFileLine(line, eventId)) {
+          return true;
+        }
+      }
+    }
+    return false;
```

Comment:
```suggestion
    boolean eventFound = false;
    if (exporterFile.length() > 0) {
      final List eventFileLines = StringUtils.split(exporterFile, EVENT_FILE_LINE_DELIMITER);
      for (int x = 1; x < eventFileLines.size(); x++) {
        String line = (String) eventFileLines.get(x);
        if (checkEventFileLine(line, eventId)) {
          eventFound = true;
          break;
        }
      }
    }
    return eventFound;
```

Not sure how cleanly this suggestion will show, but we should avoid returning in the middle of a method where possible. It makes the method more difficult to follow because there is no longer only one entry and exit point.

I suggest we add a boolean such as `boolean eventFound = false` and then `return eventFound;` at the end of the method. If we find the event during the loop, then we can do `eventFound = false; break;`

Reply:
will change